Mediapipe.Net/Calculators/GpuCalculator.cs

```diff
@@ -0,0 +1,116 @@
+
+        protected abstract string GraphPath { get; }
+        protected abstract string? SecondaryOutputStream { get; }
+
+        private readonly CalculatorGraph graph;
+        private readonly GpuResources gpuResources;
+        private readonly GlCalculatorHelper gpuHelper;
+        private readonly OutputStreamPoller<GpuBuffer> framePoller;
+        private readonly GCHandle observeStreamHandle;
+
+        [SupportedOSPlatform("Linux"), SupportedOSPlatform("Android")]
+        protected GpuCalculator()
+        {
+            graph = new CalculatorGraph(File.ReadAllText(GraphPath));
+
+            gpuResources = GpuResources.Create().Value();
+            graph.SetGpuResources(gpuResources);
+            gpuHelper = new GlCalculatorHelper();
+            gpuHelper.InitializeForTest(graph.GetGpuResources());
+
+            framePoller = graph.AddOutputStreamPoller<GpuBuffer>(output_video_stream).Value();
+
+            if (SecondaryOutputStream != null)
+            {
+                graph.ObserveOutputStream<TPacket, T>(SecondaryOutputStream, (packet) =>
+                {
+                    if (packet == null)
+                        return Status.Ok();
+
+                    T secondaryOutput = packet.Get();
+                    OnResult?.Invoke(this, secondaryOutput);
+                    return Status.Ok();
+                }, out observeStreamHandle).AssertOk();
+            }
+        }
+
+        public void Run() => graph.StartRun().AssertOk();
+
+        public ImageFrame Send(ImageFrame frame)
+        {
+            gpuHelper.RunInGlContext(() =>
+            {
+                GlTexture texture = gpuHelper.CreateSourceTexture(frame);
+                GpuBuffer gpuBuffer = texture.GetGpuBufferFrame();
+                Gl.Flush();
+                texture.Release();
+
+                var packet = new GpuBufferPacket(gpuBuffer, new Timestamp(CurrentFrame++));
+                graph.AddPacketToInputStream(input_video_stream, packet);
+
+                return Status.Ok();
+            }).AssertOk();
+
+            GpuBufferPacket outPacket = new GpuBufferPacket();
+            framePoller.Next(outPacket);
+
+            ImageFrame? outFrame = null;
+
+            gpuHelper.RunInGlContext(() =>
+            {
+                GpuBuffer outBuffer = outPacket.Get();
+                GlTexture texture = gpuHelper.CreateSourceTexture(outBuffer);
+                outFrame = new ImageFrame(outBuffer.Format.ImageFormatFor(), outBuffer.Width, outBuffer.Height, ImageFrame.GlDefaultAlignmentBoundary);
+                gpuHelper.BindFramebuffer(texture);
+                GlTextureInfo info = outBuffer.Format.GlTextureInfoFor(0);
+                Gl.ReadPixels(0, 0, texture.Width, texture.Height, info.GlFormat, info.GlType, outFrame.MutablePixelData);
+                Gl.Flush();
+                texture.Release();
+
+                return Status.Ok();
+            }).AssertOk();
```

Comment:
For some reason, right here, after it finishes running the delegate in a GL context, it tries to finalize an image frame, which I'm guessing is `outFrame`. No matter what I tell the GC, it still tries to finalize it and then fails immediately.
I don't know how to fix this issue.

Reply:
You're releasing the GL context before you even recieved a frame from what I see here.

Reply:
Found the problem. It wasn't actually here but in the `onFrame` function. The image frame that was sent to the calculator wasn't being disposed. I actually don't know why not doing that results in the delegate of that image frame being called while it's disposed somehow, but at least we know that disposing the image frame at the end of the function fixes it. Given that it's the best practice to do anyway and that it's adapted to our use-case, it's a good thing that II managed to catch it I guess.